privacyidea/lib/usercache.py

```diff
@@ -200,7 +205,7 @@ def user_init(wrapped_function, self):
         resolvers = self.get_ordererd_resolvers()
     for resolvername in resolvers:
         # If we could figure out a resolver, we can query the user cache
-        filter_conditions = create_filter(username=self.login, resolver=resolvername)
+        filter_conditions = create_filter(used_login=self.used_login, resolver=resolvername)
         result = retrieve_latest_entry(filter_conditions)
         if result:
             # Cached user exists, retrieve information and exit early
```

Comment:
We need to set ``self.login`` to ``result.username`` here. 
Otherwise, the user object will have ``login`` set to the ``used_login`` if the user was found in the cache.

Reply:
thx! Done in c1ae148c

Reply:
:+1: